resources/assets/js/helpers/router.ts

```diff
@@ -36,21 +38,33 @@ export const useUrlHash = (): void => {
   }, [location.hash, hashFound]);
 };
 
-export const useRouter = (
-  routes: Routes<any, React.ReactElement>,
-): React.ReactElement | null => {
+export const UseRouter: FunctionComponent<{
+  routes: Routes<any, any>;
+  scrollToTop?: boolean;
+}> = ({ routes, scrollToTop }): React.ReactElement | null => {
+  const intl = useIntl();
   const location = useLocation();
   const router = useMemo(() => new UniversalRouter(routes), [routes]);
   const [component, setComponent] = useState<React.ReactElement | null>(null);
+  const tracker: HTMLElement | null = document.getElementById(
+    "job-builder-root",
+  );
```

Comment:
This is specific to the Job Builder, and breaks this hook/component's potential reusability.